src/store/modules/wallet.js

```diff
@@ -472,75 +511,88 @@ const actions = {
-        collectEvents[index].price = price;
-        collectEvents[index].granterMemo = granterMemo;
-        collectEvents[index].sender = sellerWallet;
-      });
-    });
-    categorizedEvents.forEach(event => {
-      if (event.type === 'collect') {
-        Object.assign(
-          event,
-          collectEvents.find(collectEvent => collectEvent === event)
-        );
-      }
-    });
-
     // filter available events
     const filteredEvents = categorizedEvents.filter(event => {
-      if (event.sender === address || event.receiver === address) {
+      if (event.sender === address) {
         return false;
       }
-      if (event.type === 'send' || event.type === 'publish') {
-        return !!event.memo;
+      if (event.type !== 'send' && event.receiver === address) {
+        return false;
       }
-      if (event.type === 'collect' || event.type === 'purchase') {
-        return !!event.granterMemo;
+      if (event.type === 'send') {
+        return !!event.memo;
       }
-      return false;
+      return true;
     });
 
-    commit(WALLET_SET_FOLLOWEE_EVENTS, filteredEvents);
-    commit(WALLET_SET_FOLLOWEE_EVENT_FETCHING, false);
+    if (filteredEvents?.length) {
+      const existingEvents = state.followeeEvents || [];
+      const updatedEvents = [...existingEvents, ...filteredEvents];
+      commit(WALLET_SET_FOLLOWEE_EVENTS, updatedEvents);
+    }
+    return filteredEvents;
+  },
+
+  async lazyFetchEventsMemo({ commit, dispatch, getters }, categorizedEvent) {
+    if (getters.getHasFetchMemo(categorizedEvent.tx_hash))
+      return getters.getFeedEventMemo(categorizedEvent.tx_hash);
+
+    const txHash = categorizedEvent.tx_hash;
+    const event = { ...categorizedEvent };
+    let memo;
+
+    switch (event.type) {
+      case 'send':
+        memo = event.memo;
+        break;
+
+      case 'publish':
+        try {
+          const metadata = await dispatch(
+            'lazyGetNFTClassMetadata',
+            event.class_id
+          );
+          memo = metadata.message;
+        } catch (error) {
+          // eslint-disable-next-line no-console
+          console.error(error);
+          memo = '';
+        }
+        break;
+
+      case 'collect':
+      case 'purchase':
+        try {
+          const map = await getNFTHistoryDataMap({
+            axios: this.$api,
+            classId: event.class_id,
+            txHash,
+          });
+          if (map.size) {
+            map.forEach(data => {
+              const { granterMemo } = data;
+              memo = granterMemo;
+            });
+          } else {
+            memo = '';
+          }
+        } catch (error) {
+          // eslint-disable-next-line no-console
+          console.error(error);
+          memo = '';
+        }
+        break;
+
+      default:
+        memo = '';
+        break;
+    }
+
+    commit(WALLET_SET_FEED_EVENT_MEMO, {
+      txHash,
+      event: { memo },
```

Comment:
why wrapping `memo` with `event` object if there is only one key?

Reply:
I don't want to store the entire event, but I'd like to maintain some flexibility, so we can add information like `price` in the future.

Reply:
If that's the case, I suggest renaming the `WALLET_SET_FEED_EVENT_MEMO` to `WALLET_SET_FEED_EVENT_DATA`